src/Cassandra/Metadata.cs

```diff
@@ -38,6 +41,11 @@ public class Metadata : IDisposable
         private volatile TokenMap _tokenMap;
         private volatile ConcurrentDictionary<string, KeyspaceMetadata> _keyspaces = new ConcurrentDictionary<string,KeyspaceMetadata>();
         private volatile SchemaParser _schemaParser;
+        private volatile ConcurrentDictionary<string, long> _keyspacesLastUpdateTicks = new ConcurrentDictionary<string, long>();
+        private long _lastRebuildTicks = 0;
+        private readonly TaskFactory _tokenMapTaskFactory = new TaskFactory(
+            CancellationToken.None, TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning,
```

Comment:
My intention was exactly that, to avoid using tasks from the shared `ThreadPool` since this is not an asynchronous task but maybe my reasoning doesn't make much sense...

Reply:
Removed `TaskCreationOptions.LongRunning`